src/tensor.cpp

```diff
@@ -193,13 +205,68 @@ TensorDescriptor::TensorDescriptor(miopenDataType_t t,
 {
 }
 
+TensorDescriptor::TensorDescriptor(miopenDataType_t t,
+                                   miopenTensorLayout_t layout_in,
+                                   std::vector<std::size_t>&& lens_in,
+                                   std::vector<std::size_t>&& strides_in)
+    : TensorDescriptor(t, layout_in, std::move(lens_in), std::move(strides_in), true)
+{
+}
+
 // Main private constructor
 TensorDescriptor::TensorDescriptor(miopenDataType_t t,
                                    miopenTensorLayout_t layout_in,
                                    const std::vector<std::size_t>& lens_in,
                                    const std::vector<std::size_t>& strides_in,
                                    bool use_strides)
     : lens(lens_in), type(t), tensorLayout(layout_in)
+{
+    CheckArguments(t, layout_in, lens, strides_in, use_strides);
```

Comment:
two constructors are similar, so we can pick common code into an `init` function. May be `CheckArgsAndInit` and you can use `std::forward` for strides since they can be l-value or r-value.

Reply:
What about sharing like this? 

```cpp
TensorDescriptor::TensorDescriptor(miopenDataType_t t,
                                   miopenTensorLayout_t layout_in,
                                   const std::vector<std::size_t>& lens_in,
                                   const std::vector<std::size_t>& strides_in,
                                   bool use_strides)
    : lens(lens_in), strides(use_strides ? strides_in : std::vector<std::size_t>()), type(t), tensorLayout(layout_in)
{
    CheckArgsAndInit(use_strides);
}

TensorDescriptor::TensorDescriptor(miopenDataType_t t,
                                   miopenTensorLayout_t layout_in,
                                   std::vector<std::size_t>&& lens_in,
                                   std::vector<std::size_t>&& strides_in,
                                   bool use_strides)
    : lens(std::move(lens_in)), strides(use_strides ? std::move(strides_in) : std::vector<std::size_t>()), type(t), tensorLayout(layout_in)
{
    CheckArgsAndInit(use_strides);
}
```